From bc1b6fc8322b9c3bd42140421479ade0b8657c34 Mon Sep 17 00:00:00 2001
From: mal359 <mal359@icloud.com>
Date: Thu, 15 Oct 2026 20:05:00 -0400
Subject: [PATCH 4/4] xpm-nox-4.2.0-mingw-remove-path

xpm-nox: fix #8548 XPM writes invalid X Pixmap with path as C identifier

The Windows path characters : and \ were not filtered out when writing
the C identifier that is part of the XPM image.
Besides that writing the complete path inside an image can
lead to a C identifier which is too long.

So let's just remove the path and only leave the filename.
---
 libXpm-3.5.19/src/WrFFrI.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/libXpm-3.5.19/src/WrFFrI.c b/libXpm-3.5.19/src/WrFFrI.c
index cabce08..f9fdbda 100644
--- a/libXpm-3.5.19/src/WrFFrI.c
+++ b/libXpm-3.5.19/src/WrFFrI.c
@@ -119,10 +119,20 @@ XpmWriteFileFromXpmImage(
 #ifdef VMS
 	name = filename;
 #else
-	if (!(name = strrchr(filename, '/')))
+	if (!(name = strrchr(filename, '/'))
+#ifdef _WIN32
+	    && !(name = rindex(filename, ':'))
+#endif
+	)
 	    name = filename;
 	else
 	    name++;
+#ifdef _WIN32
+	if (s = rindex(name, '\\')) {
+        s++;
+        name = s;
+    }
+#endif
 #endif
 	/* let's try to make a valid C syntax name */
 	if (strchr(name, '.')) {
-- 
2.53.0.697.g625c4fb2daa

